File: 0482-License-Key-Formatting/0482.py
class Solution:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        index = len(s) - 1
        count = 0
        ans = ''

        while index >= 0:
            if s[index] == '-':
                index -= 1
                continue

            if count == k:
                ans = '-' + ans
                count = 0

            ans = s[index].upper() + ans
            count += 1

            index -= 1
        
        return ans

File: 0482-License-Key-Formatting/0482.java
class Solution {
    public String licenseKeyFormatting(String s, int k) {
        char[] arrayS = s.toCharArray();
        int index = arrayS.length - 1;
        int count = 0;
        String ans = "";

        while (index >= 0) {
            if (arrayS[index] == '-') {
                index--;
                continue;
            }

            if (count == k) {
                ans = '-' + ans;
                count = 0;
            }

            ans = Character.toUpperCase(arrayS[index]) + ans;
            count++;
            index--;
        }

        return ans;
    }
}